### PR TITLE
ci: Adding gosec in golangci lint

### PR DESCRIPTION
**Pull request by subhamkrai**
# Describe what this PR does #

ci: add gosec in golangci lint check

adding gosec in golangci lint check, also skip gosec G204 as we don't
want to run gosec on exec method

**Checklist:**

* [ ] **Commit Message Formatting**: Commit titles and messages follow
  guidelines in the [developer
  guide](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#commit-messages).
* [ ] Reviewed the developer guide on [Submitting a Pull
  Request](https://github.com/ceph/ceph-csi-operator/blob/devel/docs/development-guide.md#development-workflow)
* [ ] [Pending release
  notes](https://github.com/ceph/ceph-csi-operator/blob/devel/PendingReleaseNotes.md)
  updated with breaking and/or notable changes for the next major release.
* [ ] Documentation has been updated, if necessary.
* [ ] Unit tests have been added, if necessary.
* [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by nb-ohad** [comment]
@subhamkrai Can you please expend on the reason to disable G204?

**Comment by subhamkrai** [comment]
> @subhamkrai Can you please expend on the reason to disable G204?

@nb-ohad gosec G204 gives error when subprocess is started with in the a process, like `exec.Command` and in other repos also upstream I have seen they suggesting to either disable gosec on that line or disable `G204`

**Comment by Madhu-1** [comment]
@subhamkrai we should not disable the security rules in the whole repo, if required disable it by adding a comment where it is used or disable it in the test files (if possible)

**Comment by subhamkrai** [comment]
> @subhamkrai we should not disable the security rules in the whole repo, if required disable it by adding a comment where it is used or disable it in the test files (if possible)

we can disable for single line, but this rule was something I noticed disabled most places

**Comment by Madhu-1** [comment]
> > @subhamkrai we should not disable the security rules in the whole repo, if required disable it by adding a comment where it is used or disable it in the test files (if possible)
> 
> we can disable for single line, but this rule was something I noticed disabled most places

these kind of rules need to be verified and disable on demand based on the impact, For now lets not disable it by default rather disable it by putting a comment on the line in code

**Comment by subhamkrai** [comment]
> > > @subhamkrai we should not disable the security rules in the whole repo, if required disable it by adding a comment where it is used or disable it in the test files (if possible)
> > 
> > 
> > we can disable for single line, but this rule was something I noticed disabled most places
> 
> these kind of rules need to be verified and disable on demand based on the impact, For now lets not disable it by default rather disable it by putting a comment on the line in code

there is small problem now, we need to put `//nolint:gosec` in the same line as `exec.command` but when `exec.command` is spilt into multiple lines is failing with G204  but if we keep `exec.command` golangci lint in same long line it will fail again due to char limit per line again golangci lint is failing

**Comment by subhamkrai** [comment]
> > > > @subhamkrai we should not disable the security rules in the whole repo, if required disable it by adding a comment where it is used or disable it in the test files (if possible)
> > > 
> > > 
> > > we can disable for single line, but this rule was something I noticed disabled most places
> > 
> > 
> > these kind of rules need to be verified and disable on demand based on the impact, For now lets not disable it by default rather disable it by putting a comment on the line in code
> 
> there is small problem now, we need to put `//nolint:gosec` in the same line as `exec.command` but when `exec.command` is spilt into multiple lines is failing with G204 but if we keep `exec.command` golangci lint in same long line it will fail again due to char limit per line again golangci lint is failing

@Madhu-1 ^

**Comment by nb-ohad** [comment]
@subhamkrai Where exactly do we have `exec.command` in the repo? We should not have it in code that runs on production. Do we have something like that in tests or supporting code?

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ var _ = Describe("controller", Ordered, func() {
 		Expect(utils.InstallCertManager()).To(Succeed())
 
 		By("creating manager namespace")
-		cmd := exec.Command("kubectl", "create", "ns", namespace)
+		cmd := exec.Command("kubectl", "create", "ns", namespace) //nolint:gosec
```
Comment (subhamkrai) [inline test/e2e/e2e_test.go:41]: @nb-ohad for example this place

Comment (nb-ohad) [reply test/e2e/e2e_test.go:41]: Yes, but this is test code, not code that will run as part of a csi deployment. If all of the cases are like that then we should not diable the rule globally just locally. I believe the last push did exactly that.

**Comment by nb-ohad** [comment]
@subhamkrai Can you please fix the lint errors?

**Comment by subhamkrai** [comment]
> @subhamkrai Can you please fix the lint errors?

@nb-ohad misspell error will go away, and for golangCI lint we need to decided how we want to fix as I mentioned 
```
there is small problem now, we need to put //nolint:gosec in the same line as exec.command but when exec.command is spilt into multiple lines is failing with G204 but if we keep exec.command golangci lint in same long line it will fail again due to char limit per line again golangci lint is failing
```
in above comment

**Comment by Madhu-1** [comment]
@subhamkrai did you try to add that comment on top of that line?

File: test/utils/utils.go [hunk 2]
```diff
@@ -87,12 +90,14 @@ func UninstallCertManager() {
...
 func InstallCertManager() error {
 	url := fmt.Sprintf(certmanagerURLTmpl, certmanagerVersion)
+	//nolint:gosec
 	cmd := exec.Command("kubectl", "apply", "-f", url)
 	if _, err := Run(cmd); err != nil {
 		return err
 	}
 	// Wait for cert-manager-webhook to be ready, which can take time if cert-manager
-	// was re-installed after uninstalling on a cluster.
+	// was re-installed after uninstalling on a cluster.\
```
Comment (Madhu-1) [inline test/utils/utils.go:99]: do we need `\` at the end of this line?

Comment (subhamkrai) [reply test/utils/utils.go]: removed

**Review by Madhu-1 — APPROVED** [review]

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**